tools/interactive/interactivetool_jupyter_notebook.xml

```diff
@@ -17,12 +17,22 @@
     </environment_variables>
     <command detect_errors="aggressive"><![CDATA[
         #import re
+        #import random
+        #import string
         export GALAXY_WORKING_DIR=`pwd` &&
         mkdir -p ./jupyter/outputs/ &&
         mkdir -p ./jupyter/data &&
 
-        #set $cleaned_name = re.sub('[^\w\-\.]', '_', str($input.element_identifier))
-        ln -sf '$input' './jupyter/data/${cleaned_name}' &&
+        #set $cleaned_names = []
+        #for $file in $input
+            #set $cleaned_name = re.sub('[^\w\-\.]', '_', str($file.element_identifier))
+            ## prevent overwriting data with a randomized suffix
+            #if $cleaned_name in $cleaned_names:
+                #set $cleaned_name = $cleaned_name + '_' + ''.join(random.choice(string.ascii_lowercase) for _ in range(3))
+            #end if
+            #set $cleaned_names += [$cleaned_name]
+            ln -sf '$file' './jupyter/data/${cleaned_name}' &&
```

Comment:
You might want to add the extension as well
```suggestion
            ln -sf '$file' './jupyter/data/${cleaned_name}.${file.ext}' && 
```

Reply:
I like that idea.